mongodb/src/main/scala/quasar/physical/mongodb/fs/package.scala

```
@@ -50,48 +53,20 @@ package object fs {
+  type PhysFsEff[A]  = Coproduct[Task, PhysErr, A]
 
-    (
-      runM(WorkflowExecutor.mongoDb)                             |@|
-      queryfile.run[BsonCursor, S](client, defDb).liftM[EnvErrT] |@|
-      readfile.run[S](client).liftM[EnvErrT]                     |@|
-      writefile.run[S](client).liftM[EnvErrT]                    |@|
-      managefile.run[S](client).liftM[EnvErrT]
-    )((execMongo, qfile, rfile, wfile, mfile) => {
-      interpretBackendEffect[Free[S, ?]](
-        Empty.analyze[Free[S, ?]], // old mongo, will be removed
-        qfile compose queryfile.interpret(execMongo),
-        rfile compose readfile.interpret,
-        wfile compose writefile.interpret,
-        mfile compose managefile.interpret)
-    })
-  }
+  def parseConfig(uri: ConnectionUri)
+      : DefErrT[Task, MongoConfig] =
+    (for {
+      client <- asyncClientDef[Task](uri)
+      defDb <- free.lift(findDefaultDb.run(client)).into[Task].liftM[DefErrT]
+      wfExec <- wfExec(client)
+    } yield MongoConfig(client, defDb, wfExec)).mapT(freeTaskToTask.apply)
 
-  def definition[S[_]](implicit
-    S0: Task :<: S,
-    S1: PhysErr :<: S
-  ): BackendDef[Free[S, ?]] = BackendDef.fromPF[Free[S, ?]] {
-    case (FsType, uri) =>
-      type M[A] = Free[S, A]
-      for {
-        client <- asyncClientDef[S](uri)
-        defDb  <- free.lift(findDefaultDb.run(client)).into[S].liftM[DefErrT]
-        fs     <- EitherT[M, DefinitionError, BackendEffect ~> M](free.lift(
-                    fileSystem[S](client, defDb)
-                      .leftMap(_.right[NonEmptyList[String]])
-                      .run
-                  ).into[S])
-        close  =  free.lift(Task.delay(client.close()).attempt.void).into[S]
-      } yield BackendDef.DefinitionResult[M](fs, close)
-  }
+  def compile(cfg: MongoConfig): BackendDef.DefErrT[Task, (M ~> Task, Task[Unit])] =
+    (effToTask(cfg) map (i => (
+      foldMapNT[Eff, Task](i),
+      Task.delay(cfg.client.close).void))).liftM[DefErrT]
```

Lost an `attempt` here.  Was that intentional?

---

Hmm.. the difference is propagating a Task failure (in the current code) vs swallowing it (when adding `attempt`), right? I _think_ it's ok to just propagate a failure here?